pyuploadtool/changelog/github.py

```diff
@@ -0,0 +1,156 @@
+import os
+import requests
+
+from .changelog import Changelog
+from ..logging import make_logger
+
+
+class GitHubChangelogGenerator:
+    logger = make_logger("github-changelog-generator")
+
+    def __init__(self, metadata, github_token):
+        """
+        Prepares the changelog using GitHub REST API by
+        comparing the current commit against the latest release (pre-release / stable)
+
+        :param metadata:
+        :type metadata: ReleaseMetadata
+        :param github_token: A GitHub Personal Access Token to handle GitHub API calls
+        :type github_token: str
+        """
+        self._metadata = metadata
+
+        self.github_token = github_token
+        self.github_repository = metadata.repository_slug
+        self.commit_sha = metadata.commit
+        self.headers = {
+            "Authorization": f"token {self.github_token}"
+        }
+
+    def _get_github_api_repos(self, arg):
+        """
+        Helper function to aid github api calls.
+
+        :param arg: https://docs.github.com/en/free-pro-team@latest/rest/reference/repos
+        :type arg: str
+        :return: JSON response
+        :rtype: dict
+        """
+        api_endpoint = f"https://api.github.com/repos/{self.github_repository}/{arg}"
```

Comment:
Why not use the `github` package? What you do here is more complex than letting that package handle everything. Plus, the package is used everywhere else.

Reply:
Fixed and rebased to https://github.com/TheAssassin/pyuploadtool/pull/3/commits/1e62a479f80334d66ddb075d62e25cf6abd5f900